### PR TITLE
Add json logging to tilequeue processing

### PR DESCRIPTION
**Pull request by rmarianski**
Connects to https://github.com/mapzen/tile-tasks/issues/268

A new logging abstraction was created, and the processing worker classes now all use it for logging.

I expect this to conflict with https://github.com/tilezen/tilequeue/pull/255. I'm happy to resolve any conflicts once that gets merged, since the changes I've made are all just logging based and will not have changed any of the logic.

### CODE REVIEW

**Comment by nvkelso** [comment]
> I expect this to conflict with #255. I'm happy to resolve any conflicts once that gets merged, since the changes I've made are all just logging based and will not have changed any of the logic.

👍

File: tilequeue/log.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    QUEUE_SIZES = 3
+
+
+def log_level_name(log_level):
+    return log_level.name.lower()
+
+
+def log_category_name(log_category):
+    if log_category == LogCategory.QUEUE_SIZES:
+        return 'queue-sizes'
```
Comment (nvkelso) [inline tilequeue/log.py]: we seem to use `_` everywhere else, is the `-` deliberate here?

(I just solved a related _ versus - bug in the basemap styles so sensitive to this...)

File: tilequeue/worker.py [hunk 2]
```diff
@@ -484,45 +497,33 @@ def __call__(self):
                 if tile_timestamp_millis is not None:
                     tile_timestamp_seconds = tile_timestamp_millis / 1000.0
                     time_in_queue = now - tile_timestamp_seconds
+            timing['queue'] = time_in_queue
 
             layers = metadata['layers']
             size = layers['size']
-            size_as_str = repr(size)
+            # size_as_str = repr(size)
```
Comment (nvkelso) [inline tilequeue/worker.py]: Should this be removed entirely?

File: tilequeue/log.py (hunk 1 above)
Comment (rmarianski) [reply tilequeue/log.py]: Good call. I checked how we do the tapalcatl logging, and we also prefer `_` there too. Updated in 4310ce1.

File: tilequeue/worker.py (hunk 2 above)
Comment (rmarianski) [reply tilequeue/worker.py]: Yea, agree. Removed in ad9c83d.

File: tilequeue/log.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            logging_log_level = log_level.value
+        except:
+            print >>sys.stderr, ('ERROR: code error: invalid log level: %s' %
+                                 log_level)
+            log_level_str = log_level_name(LogLevel.ERROR)
+            logging_log_level = logging.ERROR
+        try:
+            log_category_str = log_category_name(log_category)
+        except:
+            print >>sys.stderr, ('ERROR: code error: invalid log category: %s'
```
[the remark this answers is not in the document]
Comment (rmarianski) [reply tilequeue/log.py]: Makes sense, updated in aa89b44.

File: tilequeue/worker.py [hunk 4]
```diff
@@ -62,11 +64,10 @@ def __call__(self, coord, data):
                 if self.stop.is_set():
                     return True
 
-        except MemoryError:
+        except MemoryError as e:
```
[the remark this answers is not in the document]
Comment (rmarianski) [reply tilequeue/worker.py]: To me, the way it is now seems more explicit and straightforward. I'm inclined to just leave it the way it is unless there are objections.

[the remark this answers is not in the document]
Comment (iandees) [reply tilequeue/worker.py]: Yep, sounds good to me.

**Comment by rmarianski** [comment]
Rebased the changes off of master.

File: tilequeue/worker.py [hunk 5]
```diff
@@ -491,6 +502,15 @@ def __call__(self):
             coord = data['coord']
 
             start = time.time()
+
+            try:
+                self.inflight_mgr.unmark_inflight(coord)
```
Comment (zerebubuth) [inline tilequeue/worker.py]: Nice catch! Moving this before the queue job acknowledgement will mean we're less likely to leave a tile marked as in-flight when it's not.

**Pull request merged**